src/plugins/intel_cpu/src/nodes/fullyconnected.h

```diff
@@ -75,6 +77,44 @@ class FullyConnected : public Node {
     static const size_t WEIGHTS_ID = 1;
     static const size_t BIAS_ID = 2;
     dnnl::memory::data_type outputDataType;
+
+    using executorPtr = std::shared_ptr<DnnlExecutor>;
+    executorPtr execPtr = nullptr;
+    bool shouldUseConv1x1 = false;
+    impl_desc_type implementationTypeIP;
+    MemoryDescPtr weightDescIP;
+    // when weightCache does not enable(such as stream=1) brgconv weights may change due to
+    // different shapes. Weights will cache in privateWeightCache
+    // when weightCache enabled hold weight ptr reference since weightCache does not hold the
+    // reference
+    std::map<std::string, MemoryPtr> privateWeightCache;
+
+    class ExecutorInnerProduct : public DnnlExecutor {
+        public:
+            ExecutorInnerProduct(const dnnl::inner_product_forward::primitive_desc& pd);
+    };
+
+    class ExecutorConv1x1 : public DnnlExecutor {
+        public:
+            ExecutorConv1x1(const dnnl::convolution_forward::primitive_desc& pd);
+            const dnnl::memory::desc& getSrcDesc() const { return srcDesc; }
+            const dnnl::memory::desc& getWeightDesc() const { return weightDesc; }
+            const dnnl::memory::desc& getDstDesc() const { return dstDesc; }
+            impl_desc_type getImplementationType() const { return implementationType; }
```

Comment:
What about the following. Move this interface to the `DnnlExecutor` level, so that the user may check what the actual tensor layouts are required, and, if it does make sense, he can provide memory in corresponding format to avoid extra reordering, or leave the implementation the layout mismatch handling. Also that means, that in the prepare params stage, we do not actually have to downcast to check the type. All we need is to compare the weights descriptor with the existing one from the weights input weights, and perform internal reordering in case of mismatch (I mean `FullyConnected::prepareWeightMemory` call).

Reply:
Good idea, downcast is removed.